source/api/q-search.rst

```diff
@@ -0,0 +1,112 @@
+
+
+Query results parameters
+========================
+
+The following parameters can be appended to your request to format the results:
+
+
+- ``_content_type=json``: returns results in json format. If this parameter is not provided, then the results are returned in xml format.
+
+- ``sortBy``: sorts the results by different criteria (example: ``sortBy=relevance``):
+
+  - ``relevance`` (default sorting method if not provided)
+
+  - ``title`` (metadata title)
+
+  - ``changeDate`` (metadata datestamp)
+
+  - ``rating``
+
+  - ``popularity``
+
+  - ``denominatorDesc``
+
+  - ``denominatorAsc``
+
+- ``sortOrder=reverse``: Used to sort alphabetically. Note this will sort in **ASCENDING** order (eg A - Z)
+
+- ``from``, ``to``: Used to return a subset of the results, usually for pagination (example: ``from=1&to=20``)
+
+- ``fast``: Used to indicate the information to return. Possible values:
+
+  - ``index``: returns the metadata information from the Lucene index (a subset of the information). In most cases this is the best option as the retrieval of information from the Lucene index is very fast.
+
+    The fields returned are configured in the ``dumpFields`` section in https://github.com/geonetwork/core-geonetwork/blob/master/web/src/main/webapp/WEB-INF/config-lucene.xml#L107
+
+  - ``false``: returns the raw (full) metadata. This is slower as it will retrieve every metadata attribute from the database. If this parameter is not provided, it returns a minimal set of information for each record: uuid, internal id, metadata schema, create/change dates
+
+- ``buildSummary``: Returns a summary element with search facets that can be used to filter the metadata, typically used to provide quick filters (facets) on the search results page. Values:
+
+  - ``true`` (default, if the parameter is not provided). 
+
+  - ``false``: doesn’t return the summary.
+
+- ``summaryOnly``: Returns the summary (depending on the value ofthe parameter buildSummary) and results. Values:
+
+  - ``0`` (default, if the parameter is not provided). 
+
+  - Any other value returns the summary only.
+
+- ``resultType``: type of summary to return. Summaries are configured in the ``summaryTypes`` section in https://github.com/geonetwork/core-geonetwork/blob/master/web/src/main/webapp/WEB-INF/config-summary.xml#L132-L249
+
+  - ``hits`` (default value if not provided), returns the fields configured in the ``hits`` section in https://github.com/geonetwork/core-geonetwork/blob/master/web/src/main/webapp/WEB-INF/config-summary.xml#L185
+
+  - ``details`` (recommended value to send), returns the fields configured in the ``details`` section in https://github.com/geonetwork/core-geonetwork/blob/master/web/src/main/webapp/WEB-INF/config-summary.xml#L133
+
+- Other values in the summaries section are allowed
+
```

Comment:
Another useful field could be `extraDumpFields`. It is a list of Lucene fields separated by `,` that you want to be added to the response, apart of the fields defined in the `summaryType` used.
https://github.com/geonetwork/core-geonetwork/blob/8ba8e6eca3edab06adb6ffbb14422c83ef14b8d8/core/src/main/java/org/fao/geonet/kernel/search/LuceneSearcher.java#L1143-L1152